stratum/lib/macros.h

```diff
@@ -66,6 +66,16 @@ inline const std::string FixMessage(const std::string& msg) {
                   : " ")                                                       \
           << FixMessage(__ret.status().error_message())
 
+// A macro for simplifying creation of a new error or appending new info to an
+// error based on the provided error code
+#define APPEND_ERROR_WITH_CODE(out, code)                                      \
+  if (!out.ok()) assert(out.error_code() == code);                             \
```

Comment:
assert() isn't really used on any non-test codepath, so we might want to do something else here (e.g. log an error)